lib/index.ts

```diff
@@ -85,10 +85,38 @@ class PWMetrics {
         const sheets = new Sheets(this.sheets, this.clientSecret);
         await sheets.appendResults(results.runs);
       }
+
+      if (this.resultHasExpectationErrors(results) && this.flags.expectations) {
+        throw new Error(messages.getMessage('HAS_EXPECTATION_ERRORS'));
+      }
     }
     return results;
   }
 
+  resultHasExpectationErrors(results: PWMetricsResults): boolean {
+    const expectationErrors = results.runs.filter(run => {
+      return run.timings.filter(item => {
```

Comment:
this is a pretty heavyweight way to dive back into the results and expectations. Can we instead set a flag on the state of the run while `checkExpectations` runs?